InaraResponder/InaraResponder.cs

```diff
@@ -658,41 +664,47 @@ private void handleShipSwappedEvent(ShipSwappedEvent @event)
-                    { "shipName", storedShip.name },
-                    { "shipIdent", storedShip.ident },
-                    { "isHot", storedShip.hot },
-                    { "shipRole", storedShip.Role.invariantName },
-                    { "isCurrentShip", true },
+                    { "shipType", @event.storedEdModel },
+                    { "shipGameID", @event.storedshipid },
+                    { "isCurrentShip", false },
                     { "starsystemName", EDDI.Instance.CurrentStarSystem?.systemname },
                     { "stationName", EDDI.Instance.CurrentStation?.name },
                     { "marketID", EDDI.Instance.CurrentStation?.marketId }
                 };
+                var storedShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.storedshipid);
+                if (storedShip?.EDName == @event.storedEdModel)
+                {
+                    storedShipData.Add("shipName", storedShip?.name);
+                    storedShipData.Add("shipIdent", storedShip?.ident);
+                    storedShipData.Add("isHot", storedShip?.hot ?? false);
+                    storedShipData.Add("shipRole", (storedShip?.Role ?? Role.MultiPurpose).invariantName);
+                }
                 inaraService.EnqueueAPIEvent(new InaraAPIEvent(@event.timestamp, "setCommanderShip", storedShipData));
             }
             else if (!string.IsNullOrEmpty(@event.soldship))
             {
                 inaraService.EnqueueAPIEvent(new InaraAPIEvent(@event.timestamp, "delCommanderShip", new Dictionary<string, object>()
                 {
-                    { "shipType", @event.soldShipDefinition?.EDName ?? @event.soldship },
-                    { "shipGameID", @event.storedshipid }
+                    { "shipType", @event.soldEdModel },
+                    { "shipGameID", @event.soldshipid }
                 }));
             }
-            Ship currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
-            Dictionary<string, object> currentShipData = new Dictionary<string, object>()
+            var currentShipData = new Dictionary<string, object>()
             {
-                { "shipType", currentShip.EDName },
-                { "shipGameID", currentShip.LocalId },
-                { "shipName", currentShip.name },
-                { "shipIdent", currentShip.ident },
-                { "shipRole", currentShip.Role.invariantName },
-                { "isHot", currentShip.hot },
+                { "shipType", @event.edModel },
+                { "shipGameID", @event.shipid },
                 { "isCurrentShip", true }
             };
+            var currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
+            if (currentShip?.EDName == @event.edModel)
+            {
+                currentShipData.Add("shipName", currentShip?.name);
```

Comment:
We can remove the `?` from `currentShip`.

Reply:
No. The `GetShip` method can return null a null value if the local ship id is null or not found in the shipyard.